File: src/portfolio_manager/models/holding.py
"""Holding model."""

from dataclasses import dataclass
from datetime import datetime
from decimal import Decimal
from uuid import UUID


@dataclass
class Holding:
    """A holding of a stock in an account."""

    id: UUID
    account_id: UUID
    stock_id: UUID
    quantity: Decimal
    created_at: datetime
    updated_at: datetime

File: internal/models/holding.go
package models

import (
	"time"

	"github.com/kadragon/portfolio-manager/internal/numeric"
	"github.com/kadragon/portfolio-manager/internal/uuidx"
)

type Holding struct {
	ID        uuidx.UUID
	AccountID uuidx.UUID
	StockID   uuidx.UUID
	Quantity  numeric.Decimal
	CreatedAt time.Time
	UpdatedAt time.Time
}
